### PR TITLE
Add resource google_network_services_edge_cache_origin

### PR DESCRIPTION
**Pull request by ScottSuarez**
1/5 resources to help with https://github.com/hashicorp/terraform-provider-google/issues/7807

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_network_services_edge_cache_origin`(beta)
```

### CODE REVIEW

**Comment by ScottSuarez** [comment]
Tests failed due to existing bug `b/179502593`. Addressed with team they are rolling out a fix soon.

File: mmv1/products/networkservices/api.yaml [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: NetworkServices
+versions:
+  - !ruby/object:Api::Product::Version
+    name: beta
+    base_url: https://networkservices.googleapis.com/v1alpha1/
```
Comment (slevenick) [inline mmv1/products/networkservices/api.yaml:19]: The resource is beta but the endpoint is alpha?

File: mmv1/products/networkservices/api.yaml [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+          Once maxAttempts to this origin have failed the failoverOrigin will be used, if one is specified. That failoverOrigin may specify its own maxAttempts,
+          retryConditions and failoverOrigin to control its own cache fill failures.
+
+          The total number of allowed attempts to cache fill across this and failover origins is limited to four.
+          The total time allowed for cache fill attempts across this and failover origins can be controlled with maxAttemptsTimeout.
+
+          The last valid response from an origin will be returned to the client.
+          If no origin returns a valid response, an HTTP 503 will be returned to the client.
+
+          Defaults to 1. Must be a value greater than 0 and less than 4.
```
Comment (slevenick) [inline mmv1/products/networkservices/api.yaml:111]: We should add this as plan time validation via terraform.yaml

Comment (slevenick) [reply mmv1/products/networkservices/api.yaml:111]: There is an existing function `validation.IntBetween(x,y)`

File: mmv1/products/networkservices/api.yaml [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+          retryConditions apply to this origin, and not subsequent failoverOrigin(s),
+          which may specify their own retryConditions and maxAttempts.
+
+          Valid values are:
+
+          - CONNECT_FAILURE: Retry on failures connecting to origins, for example due to connection timeouts.
+          - HTTP_5XX: Retry if the origin responds with any 5xx response code, or if the origin does not respond at all, example: disconnects, reset, read timeout, connection failure, and refused streams.
+          - GATEWAY_ERROR: Similar to 5xx, but only applies to response codes 502, 503 or 504.
+          - RETRIABLE_4XX: Retry for retriable 4xx response codes, which include HTTP 409 (Conflict) and HTTP 429 (Too Many Requests)
+          - NOT_FOUND: Retry if the origin returns a HTTP 404 (Not Found). This can be useful when generating video content, and the segment is not available yet.        item_type: !ruby/object:Api::Type::Enum
```
Comment (slevenick) [inline mmv1/products/networkservices/api.yaml:139]: Extra `item_type` here?

**Review by slevenick — CHANGES_REQUESTED** [review]

**Comment by slevenick** [comment]
Oh, I'd also prefer to wait on merging until tests pass reliably. Unless that will take an unusually long amount of time

File: mmv1/products/networkservices/api.yaml (hunk 1 above)
Comment (ScottSuarez) [reply mmv1/products/networkservices/api.yaml:19]: This resource will never be in true `beta` . Will always be whitelist only. I guess I should clarify with a large disclaimer?

**Pull request closed without merging**